CReLU.lua

```diff
@@ -0,0 +1,51 @@
+local CReLU, parent = torch.class('nn.CReLU', 'nn.Sequential')
+
+-- Implements the CReLU activation function as described by
+-- W. Shang et al. in "Understanding and Improving Convolutional Neural Networks
+-- via Concatenated Rectified Linear Units"
+-- Note: Only supports batched input of the form [B,C,H,W]
+function CReLU:__init(inplace)
+   parent.__init(self)
+   local concatTable = nn.ConcatTable()
+   concatTable:add(nn.Identity())
+   concatTable:add(nn.MulConstant(-1))
+   self.inplace = inplace or false
+   self.batched = true
+   self:add(concatTable)
+   self:add(nn.JoinTable(2))
+   self:add(nn.ReLU(self.inplace))
+end
+
+
+function CReLU:updateOutput(input)
+   self.batched = true
+   if input:nDimension() == 3 then
+      self.batched = false
+   end
+
+   if self.batched then
+      parent.updateOutput(self, input)
+   else
+      parent.updateOutput(self, input:view(1, input:size(1), input:size(2), input:size(3)))
```

Comment:
@soumith  thanks for the help so far. I've made the changes to `updateOutput` to use `nInputDims` and also made the following changes:

- Changed the test to no longer check that the first `x/2` feature maps were `ReLU(input)` and the second `x/2` were `ReLU(-input)` as now the order is not guaranteed, instead I check that the output feature maps contains all of the ones from `ReLU(input)` and `ReLU(-input)`

- I added `CReLU:backward` as well as `CReLU:updateGradInput` as `CReLU` is of type `Sequential` and `Sequential.backward` doesn't call `updateGradInput`, but `Jacobian.testJacobian` does, so both need to be defined (if I'm not mistaken, please let me know if you agree)

- Updated docs to reference the `nInputDims` parameter.